src/freenet/client/filter/TheoraPacketFilter.java

```diff
@@ -1,148 +1,287 @@
+		Logger.minor(this, "IDENTIFICATION_HEADER");
+
+		// The header packets begin with the header type and the magic number. Validate both.
+		byte[] magicHeader = new byte[1 + magicNumber.length];
+		input.readFully(magicHeader);
+		checkMagicHeader(magicHeader, (byte) 0x80); // -128
+
+		int VMAJ = input.readInt(8);
+		if (VMAJ != 3) throw new UnknownContentTypeException("Header VMAJ: " + VMAJ);
+
+		int VMIN = input.readInt(8);
+		if (VMIN != 2) throw new UnknownContentTypeException("Header VMIN: " + VMIN);
+
+		int VREV = input.readInt(8);
+		if (VREV > 1) throw new UnknownContentTypeException("Header VREV: " + VREV);
+
+		int FMBW = input.readInt(16);
+		if (FMBW == 0) throw new UnknownContentTypeException("Header FMBW: " + FMBW);
+
+		int FMBH = input.readInt(16);
+		if (FMBH == 0) throw new UnknownContentTypeException("Header FMBH: " + FMBH);
+
+		int PICW = input.readInt(24);
+		if (PICW > FMBW * 16) throw new UnknownContentTypeException("Header PICW: " + PICW + "; FMBW: " + FMBW);
+
+		int PICH = input.readInt(24);
+		if (PICH > FMBH * 16) throw new UnknownContentTypeException("Header PICH: " + PICH + "; FMBH: " + FMBH);
+
+		int PICX = input.readInt(8);
+		if (PICX > FMBW * 16 - PICX) throw new UnknownContentTypeException("Header PICX: " + PICX + "; FMBW: " + FMBW + "; PICX: " + PICX);
+
+		int PICY = input.readInt(8);
+		if (PICY > FMBH * 16 - PICY) throw new UnknownContentTypeException("Header PICY: " + PICY + "; FMBH: " + FMBH + "; PICY: " + PICY);
+
+		int FRN = input.readInt(32);
+		if (FRN == 0) throw new UnknownContentTypeException("Header FRN: " + FRN);
+
+		int FRD = input.readInt(32);
+		if (FRD == 0) throw new UnknownContentTypeException("Header FRN: " + FRN);
+
+		input.skip(48); // skip PARN and PARD
+
+		int CS = input.readInt(8);
+		if (!(CS == 0 || CS == 1 || CS == 2)) throw new UnknownContentTypeException("Header CS: " + CS);
+
+		input.skip(35); // skip NOMBR, QUAL and KFGSHIFT
+
+		int PF = input.readInt(2);
+		if (PF == 1) throw new UnknownContentTypeException("Header PF: " + PF);
+
+		int Res = input.readInt(3);
+		if (Res != 0) throw new UnknownContentTypeException("Header Res: " + Res);
+
+		expectedPacket = Packet.COMMENT_HEADER;
+	}
+
+	private CodecPacket parseCommentHeader(BitInputStream input) throws IOException {
+		Logger.minor(this, "COMMENT_HEADER");
+
+		byte[] magicHeader = new byte[1 + magicNumber.length];
+		input.readFully(magicHeader);
+		checkMagicHeader(magicHeader, (byte) 0x81); // -127
+
+		int vendorLength = input.readInt(32, ByteOrder.LITTLE_ENDIAN);
+		byte[] vendor = new byte[vendorLength];
+		input.readFully(vendor);
+		Logger.minor(this, "Vendor string is: " + new String(vendor, StandardCharsets.UTF_8));
+		int numberOfComments = input.readInt(32, ByteOrder.LITTLE_ENDIAN);
+		for (long i = 0; i < numberOfComments; i++) {
+			int commentLength = input.readInt(32, ByteOrder.LITTLE_ENDIAN);
+			byte[] comment = new byte[commentLength];
```

Comment:
Hmm, this does allow a denial-of-service attack on every node trying to filter this file… maybe we should restrict these kinds of allocations to some kind of limit? I have no idea whether we already do this anywhere else… @nextgens @Thynix @ArneBab do we?

Reply:
FRED has protection on this type of bugs https://git.io/fj6zx
I also added limits here

Reply:
That is totally not what I mean. I was thinking more along the lines of…
```
if (commentLength > 16 * 1024) { // amount is negotiable
    throw new UnknownContentException("Comment too large!");
}
```
Then again, if we only read the comment for logging, just read 50 bytes to print them and skip the rest? I’m really not sure we should read the data at all if we’re not using it.